### PR TITLE
Adding register, login, logout features

### LINKED ISSUE #7: login api
please send JWT as a cookie that is httponly and secure only. Also to prevent csrf use SameSite attribute of cookie

### LINKED ISSUE #8: logout api

### PR DESCRIPTION
**Pull request by SamTheBosssss**
Backend devs (i.e. @benedict-soh ) will need to run `pip3 install -r requirements.txt` again.
Closes #7 #8

### CODE REVIEW

**Review by LittenJV — APPROVED** [review]
fantastic work!

File: flask-api/flaskr/auth.py [hunk 1]
```diff
@@ -40,4 +47,43 @@ def register():
...
+
+@bp.route('/logout', methods=['POST', 'GET'])
+def logout():
+    resp = jsonify({'logout': True})
+    unset_jwt_cookies(resp)
+    return resp, 200
+
+
+@bp.route('/test', methods=['POST', 'GET'])
+@jwt_required
```
Comment (SamTheBosssss) [inline flask-api/flaskr/auth.py:87]: @benedict-soh this is all u need else where for JWT validation. Just dont forget to import the package.

**Review by a1719756 — APPROVED** [review]
Looks good.

File: flask-api/flaskr/auth.py [hunk 2]
```diff
@@ -40,4 +47,43 @@ def register():
     )
     db.commit()
     # return redirect(url_for('auth.login'))
-    return("good shit")
+    return "good shit"
```
Comment (benedict-soh) [inline flask-api/flaskr/auth.py:50]: can we return something better than "good shit" lol

File: flask-api/requirements.txt [hunk 3]
```diff
@@ -1,7 +1,10 @@
 click==7.1.2
 Flask==1.1.2
+Flask-JWT-Extended==3.24.1
 itsdangerous==1.1.0
 Jinja2==2.11.2
 MarkupSafe==1.1.1
 pkg-resources==0.0.0
+PyJWT==1.7.1
+six==1.15.0
 Werkzeug==1.0.1
```
Comment (benedict-soh) [inline flask-api/requirements.txt:10]: do I need to run the command again to update the stuff?

**Review by benedict-soh — APPROVED** [review]

**Pull request merged**